persistence/relational-jdbc/src/main/java/org/apache/polaris/persistence/relational/jdbc/JdbcMetaStoreManagerFactory.java

```diff
@@ -212,7 +211,7 @@ public synchronized Supplier<BasePersistence> getOrCreateSessionSupplier(
       checkPolarisServiceBootstrappedForRealm(
           realmContext, metaStoreManagerMap.get(realmContext.getRealmIdentifier()));
     }
-    return sessionSupplierMap.get(realmContext.getRealmIdentifier());
+    return sessionSupplierMap.get(realmContext.getRealmIdentifier()).get();
```

Comment:
nit: previously the `.get()` call was outside the `synchronized` block... not sure it matters, though :thinking: